File: chapter-17/springboot-kotlin-demo/src/main/kotlin/com/apress/demo/HomeController.kt
package com.apress.demo


import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class HomeController(val repository:UserRepository) {

    @GetMapping("/")
    fun home(model: Model): String {
        model.addAttribute("users", repository.findAll())
        return "home"
    }
}

File: chapter-17/springboot-scala-demo/src/main/scala/com/apress/demo/HomeController.scala
package com.apress.demo

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.{GetMapping}

@Controller
class HomeController
{
  @Autowired
  var repo: UserRepository = _

  @GetMapping(Array("/"))
  def home(model: Model) : String = {
    model.addAttribute("users", repo.findAll())
    "home"
  }
}
